### DISCUSSION TITLE
What are valid values for "failure_capture recover" and how should I be managing code failures?

### DISCUSSION THREAD

**Discussion by bobsalko** [snl-dakota/dakota#115] [Q&A] [answered]
I am attempting to perform a sensitivity analysis on my code.  I am trying to use variance_based_decomp.  I have everything set up and am able to run, but there are some combinations of input parameters that cause my code to fail.  Ideally, I would just like to skip these failing cases and avoid that region of the design space.  It sounds like I should use `failure_capture recover` to do this, which requires dummy function values.  My question is, what are valid values for these dummy values?  Should they be something reasonable or something unreasonable?  Will they be used at all in determining statistics for the sensitivity analysis?  Or should I be using some other means of failure detection completely?

**Comment by briadam** [comment, marked as answer]
Hi Bob -- For sampling (and select other) methods, any function values with a returned value of "NaN" or "Inf" will be omitted from statistics computation. I would recommend returning one of those strings in your analysis driver for your function value for cases where you code fails. This effectively means that you are deviating from the original specified input probability distribution, but often doesn't make a huge difference for sensitivity analysis (assuming a small fraction of samples fail to run). -- Brian

**Comment by bobsalko** [comment]
Brian, as a follow up question to this one, I ran my 1,000 sample case returning NaN for any cases that fail.  Out of 38,000 individual cases run, I have 935 failures.  The strange thing is that the Dakota output file has nothing listed for the Sobol indices.  It is just blank for each of the figures of merit I selected.  When I ran a smaller case with only 10 samples (380 simulations), I did get Sobol indices.  I don't see any error in the output file.  Any idea why Dakota would just not calculate the indices at the end?
